### ISSUE TITLE
`PATH` different between `docker exec` and `docker run` contexts

### ISSUE THREAD

**Issue by MattF-NSIDC** [mamba-org/micromamba-docker#233]
I have read #128 but would like to request an officially-documented method to `docker exec` into a micromamba container.

My use case is installing a single environment into a Docker container which will run a long-running service (`luigi`) to which processing jobs can be submitted by CLI, accessed via `docker exec`. In the past I've leveraged the `continuumio/miniconda3` image to achieve this, but it required hacks in my Dockerfile to successfully activate the non-base environment. I hoped to achieve the same without activation hacks using `micromamba` to install only my desired dependencies into a base environment.

When I try to achieve this with the micromamba docker image, I'm observing that the environment is activated and `PATH` is set correctly when using `docker run`, but not when using `docker exec` (unless I do a workaround detailed below).

### Good `PATH` with `docker run`:

```
$ docker run --rm mambaorg/micromamba:1.0 env | grep PATH
PATH=/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

### Env not activated with `docker exec`:

```
$ docker run --rm -d mambaorg/micromamba:1.0 sleep 120
ac14874f4f366091050bc419a2609f11a543455c5e59e7861c274cd8b17cca0e
$ docker exec ac14874f4f366091050bc419a2609f11a543455c5e59e7861c274cd8b17cca0e env | grep PATH
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

### Workaround to activate with `docker exec`

```
$ docker exec -t ac14874f4f366091050bc419a2609f11a543455c5e59e7861c274cd8b17cca0e bash -i -c "env" | grep PATH
PATH=/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

Maybe less effective/advisable method?: I found that adding this line to my Dockerfile based on `mambaorg/micromamba:1.0` resulted in a good-looking `PATH`, but I don't know that this is the same as correctly "activating" the base environment.

```
ENV PATH "$MAMBA_ROOT_PREFIX/bin:$PATH"
```

## Request for information

If there is no known way to resolve this problem (per #128 ), is the recommended method to achieve this `docker exec -t <container> bash -i -c "command"`? I'd be happy to open a docs PR for this once confirmed.

Thank you for your time!

**Comment by wholtz** [comment]
Hi @MattF-NSIDC thank you for your interest in micromamba and your detailed issue.

I just reviewed #128 and the information in there is still correct. My official recommendation is to do:

```
docker exec <container_id> micromamba run -n <environment_name> <command>
```

If you want to use the `base` environment, you can leave out `-n <environment_name>`.

Unfortunately, I don't see how to achieve a more automatic method given how `docker exec` works.

This should be covered in the docs. A PR would be appreciated. Or I'll get to it in a week or two. Thanks again!

**Comment by maresb** [comment]
> `ENV PATH "$MAMBA_ROOT_PREFIX/bin:$PATH"`

Technically that is not correct activation, but in practice it works flawlessly about 98% of the time.

**Comment by MattF-NSIDC** [comment]
Thanks all :)

I'll try this week to throw together a write-up "on `docker exec`" which covers the "why is this a problem" (`docker exec` ignores entrypoint and runs command in a non-interactive, non-login shell) and "how to work around the problem" (Emphasized as recommended: `docker exec ... micromamba run ...`; not recommended and non-mamba-specific: `docker exec ... bash -i -c "..."`; not recommended and may not always work, but configurable at build-time and requires no special knowledge by the person running exec: `ENV PATH "$MAMBA_ROOT_PREFIX/bin:$PATH"`).

**Comment by maresb** [comment]
@MattF-NSIDC, I'm not sure why I didn't think of this sooner, but I think the best workaround may be to explicitly invoke the current entrypoint wrapper, i.e.

```bash
docker exec -t <container_id> _entrypoint.sh env | grep PATH
```

**Comment by MattF-NSIDC** [comment]
Happy to make that change. Do you think that should be the officially-recommended method?

**Comment by wholtz** [comment]
Somehow I didn't get a notification for the last two comments here and I merged the PR before seeing them.  Had I see these messages I would have held off until the discussion was further resolved.

I've debated this in my head for a couple of minutes now. I like the idea of re-using `_entrypoint.sh` to take advantage of the checks that are embedded within that script.  But using `micromamba run` has a robustness advantage that I feel is larger.  If an image derived from `mambaorg/micromamba` were to move or modify `_entrypoint.sh`, then calling `_entrypoint.sh` could fail, but `micromamba run` should still work. 

If `docker exec <container_id> _entrypoint.sh` were used by someone, I'd recommend supplying the absolute path: `/usr/local/bin/_entrypoint.sh` for a bit of extra safety.
